### PR TITLE
DDCE-4964 Create new views, routes and use session keys for new limit…

### PR DESCRIPTION
**Pull request by mywyau**
… exceeded content

So summary for pr:

Passengers has essentially a running list of products which can be added to in the json. There is also a json field called workingInstances which pulls the item you wish to edit using it's item id called (iid). We use this to help us keep track of the total and on an edit calculate if we need to go to the overlimit kickout page and display the correct total to the user

Also done some refactoring to try and clean up the tobacco input controller and alcohol controller. moved the forms into individual form classes and respective tests.

Note The frontend cache repo only has 2 main ways of data manip, fetching and updating. Old team did create methods of removing and clearing down the product list but the methods need cleaning up and all to be moved to a single service. They were and still are marked with Todos.

There were also some issues surrounding rounding and decimal places, especially during conversion between kilos and grams combined with 0.01 (2 decimal place) value entries.

tl;dr

- tried to refactor forms/controllers as it was hard to manipulate/structure/add logic.
- changed decimal formatters as it was causing havoc with calculations and rounding.
- updated view and content.
- Added view content util for helping generate all the content needed.
- updated controller logic
- try to add tests for coverage
- added helpers to try and make calculations easier.

### CODE REVIEW

**Review by gastonracui — CHANGES_REQUESTED** [review]

File: app/util.scala [hunk 1]
```diff
@@ -78,18 +67,21 @@ package object util {
     }
   }
 
-  def calculatorLimitConstraintOptionInt(
-    limits: Map[String, BigDecimal] = Map.empty,
-    applicableLimits: List[String] = Nil
-  ): Boolean = {
+  def cigaretteAndHeatedTobaccoConstraint(numberOfSticks: Int): Boolean = numberOfSticks <= 800
```
[the remark this answers is not in the document]
Comment (mywyau) [reply app/util.scala:70]: done

File: app/util.scala [hunk 2]
```diff
@@ -78,18 +67,21 @@ package object util {
...
 
-    val errors =
-      for (limit <- applicableLimits; amount <- limits.get(limit) if amount > BigDecimal(1.0)) yield (limit, amount)
+  def cigarAndCigarilloConstraint(numberOfSticks: Int, productType: String): Boolean =
+    productType match {
+      case "cigars"     => numberOfSticks <= 200
+      case "cigarillos" => numberOfSticks <= 400
```
[the remark this answers is not in the document]
Comment (mywyau) [reply app/util.scala:75]: done

File: app/utils/FormatsAndConversions.scala [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      override def toBigDecimal(value: Int): BigDecimal = BigDecimal(value)
+    }
+
+  implicit class CovertToBigDecimal[A](value: A)(implicit tc: ToBigDecimal[A]) {
+    def toBigDecimal: BigDecimal =
+      tc.toBigDecimal(value)
+  }
+
+  implicit class BigDecimalHelper(value: BigDecimal) {
+    def format2dps: BigDecimal =
```
[the remark this answers is not in the document]
Comment (mywyau) [reply app/utils/FormatsAndConversions.scala:43]: done

File: app/views/LimitExceededViewUtils.scala [hunk 4]
```diff
@@ -0,0 +1,421 @@
...
+      case name if name.contains("cigarillos") => stickTobacco
+      case name if name.contains("heated")     => stickTobacco
+      case name if name.contains("chewing")    => looseTobacco
+      case name if name.contains("rolling")    => looseTobacco
+      case _                                   => None
+    }
+
+  private[views] def determineSinglularOrPlural[A](amount: String, singular: A, plural: A) =
+    amount match {
+      case "1.00" | "1" | "1.000" => singular
```
[the remark this answers is not in the document]
Comment (mywyau) [reply app/views/LimitExceededViewUtils.scala:44]: done

File: app/controllers/LimitExceedController.scala [hunk 5]
```diff
@@ -19,35 +19,265 @@ package controllers
...
+        val totaledAmountFormatted: BigDecimal = totaledAmount.format3dps
+
+        userInput match {
+          case Some(_) =>
+            Future(
+              Ok(
+                limitExceedViewEdit(
+                  totalEnteredAmount = totaledAmountFormatted.toString,
+                  originalAmountEntered = originalAmountFormatted.toString,
+                  userInput = userInputBigDecimalFormatted.toString(),
```
[the remark this answers is not in the document]
Comment (mywyau) [reply app/controllers/LimitExceedController.scala]: done

**Review by gastonracui — APPROVED** [review]
